modules/repository-azure/build.gradle

```diff
@@ -193,56 +185,44 @@ if (!azureAccount && !azureKey && !azureContainer && !azureBasePath && !azureSas
   azureBasePath = ''
   azureSasToken = ''
   useFixture = true
-
-  apply plugin: 'elasticsearch.test.fixtures'
-  testFixtures.useFixture ':test:fixtures:azure-fixture', 'azure-fixture'
 }
 
 Map<String, String> expansions = [
   'container': azureContainer,
   'base_path': azureBasePath + "_integration_tests"
 ]
 
-tasks.named("processYamlRestTestResources").configure {
+tasks.named("processYamlRestTestResources") {
   inputs.properties(expansions)
   filter("tokens" : expansions, ReplaceTokens.class)
 }
 
-tasks.named("internalClusterTest").configure {
+tasks.named("internalClusterTest") {
   // this is tested explicitly in a separate test task
   exclude '**/AzureStorageCleanupThirdPartyTests.class'
 }
 
-testClusters.matching { it.name == "yamlRestTest" }.configureEach {
-    keystore 'azure.client.integration_test.account', azureAccount
-    if (azureKey != null && azureKey.isEmpty() == false) {
-      keystore 'azure.client.integration_test.key', azureKey
-    }
-    if (azureSasToken != null && azureSasToken.isEmpty() == false) {
-      keystore 'azure.client.integration_test.sas_token', azureSasToken
-    }
-    if (useFixture) {
-      setting 'azure.client.integration_test.endpoint_suffix', azureAddress
-
-      def ignoreTestSeed = providers.systemProperty('ignore.tests.seed').isPresent() ? PropertyNormalization.IGNORE_VALUE : PropertyNormalization.DEFAULT
-      setting 'thread_pool.repository_azure.max', (Math.abs(BuildParams.random.nextLong() % 10) + 1).toString(), ignoreTestSeed
-    }
+tasks.named("yamlRestTest") {
+  systemProperty 'test.azure.fixture', Boolean.toString(useFixture)
+  systemProperty 'test.azure.account', azureAccount
+  systemProperty 'test.azure.container', azureContainer
+  systemProperty 'test.azure.key', azureKey
+  systemProperty 'test.azure.sas_token', azureSasToken
 }
 
 tasks.register("azureThirdPartyTest", Test) {
   SourceSetContainer sourceSets = project.getExtensions().getByType(SourceSetContainer.class);
   SourceSet internalTestSourceSet = sourceSets.getByName(InternalClusterTestPlugin.SOURCE_SET_NAME)
   setTestClassesDirs(internalTestSourceSet.getOutput().getClassesDirs())
   setClasspath(internalTestSourceSet.getRuntimeClasspath())
-  dependsOn "internalClusterTest"
+  // TODO: I think this should probably depend on 'yamlRestTest' given that task supports using a real azure endpoint
```

Comment:
I believe this `dependsOn "internalClusterTest"` is actually a mistake, rather I think we want to run the YAML REST tests as those are wired up to support running against a "real" Azure object store.

Reply:
I've rewired this so that we run the `yamlRestTest` task against a real azure service in our third party tests which I believe is the original intention. I tested this locally and confirmed it works as well.